src/main/scala/akka/rabbitmq/ChannelActor.scala

```diff
@@ -105,74 +105,85 @@ class ChannelActor(setupChannel: (Channel, ActorRef) => Any)
 
     case Event(_: ShutdownSignal, _) => stay()
   }
+
   when(Connected) {
     case Event(newChannel: Channel, Connected(channel)) =>
       log.debug("{} closing unexpected channel {}", header(Connected, newChannel), channel)
-      closeIfOpen(channel)
+      close(channel)
       stay using Connected(setup(newChannel))
 
-    case Event(msg: ShutdownSignal, Connected(channel)) =>
-      if (shutdownSignalAppliesToChannel(msg, channel)) {
-        log.debug("{} shutdown", header(Connected, msg))
-        dropChannelAndRequestNewChannel(channel)
+    case Event(shutdownSignal: ShutdownSignal, Connected(channel)) =>
+      (shutdownSignal match {
+        case ParentShutdownSignal =>
+          Some(dropChannel _) // The parent is responsible for providing the new channel.
```

Comment:
Fixes a problem where both the parent (connection actor) and the child (channel actor) request a new channel. A test has been added to check it is now happening only once, by the parent.